woocommerce/woocommerce-ios
# Product Creation AI: Generate product info

Closes: #10773 
Closes: #10771  

## Description

Identify language, generate product details and set the generated product into the view model. 

#### Changes

- Make shipping values optional and skip asking AI about it if dimension and weight units are unavailable.
- Set `Product` model’s `regularPrice` property instead of `price`.
- Set a default AITone if not already set. 
- Finetune AI prompt's price generation part. Internal - p1695773759702609-slack-C03L1NF1EA3
- Identify language and generate product.
- Add/update unit tests.


## Testing instructions

- Create a JN site with the Jetpack AI plugin. (Checkbox available on JN home page to install this plugin)
- Login into the app using the JN site
- Switch to the Products tab and select the "+" button to add a new product. 
- Tap "Create a product with AI" option
- Enter a product name and tap "Continue"
- Enter product features in the next screen and tap "Create Product Details"
- Optionally tap on "Set tone and voice" and select a tone.
- The product details should be displayed on the next screen after the loading animation.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/524475/fa74346d-872c-4143-a692-8d54d1e56ec8



---
